lms/envs/common.py

```diff
@@ -3129,6 +3129,9 @@ def _make_locale_paths(settings):  # pylint: disable=missing-function-docstring
     # in the LMS process at the moment, so anything that has Django admin access
     # permissions needs to be listed as an LMS app or the script will fail.
     'user_tasks',
+
+    # XBlocks
+    'drag_and_drop_v2',
```

Comment:
@eLRuLL I know this is the lynchpin of your solution to this issue, but we can't do this. ☹️ 

XBlocks are deliberately not fully-fledged Django applications, and have to interact with the LMS at the backend via the runtime service objects provided to them, or at the frontend via the REST APIs. They're not allowed to just import any module they want to, e.g. to query user state or the database directly.

As a small example, django applications can override translations in the main LMS, and so an XBlock added to this list could override translation strings for areas of the site which are outside of the XBlock's scope.

The whole motivation for https://github.com/edx/xblock-utils/pull/48 was to allow XBlock translations to be used within the XBlock alone, via the `i18n` runtime service.

So we have to find another way to deal with this issue. I'm sorry!

Reply:
@pomegranited yeah and that's exactly the only good xblock example that I was able to find that is working correctly

Reply:
@eLRuLL  I found the issue, and here's a way to fix it without adding this change, and without https://github.com/edx-solutions/xblock-drag-and-drop-v2/pull/266:

```diff
diff --git a/openedx/core/djangoapps/xblock/runtime/runtime.py b/openedx/core/djangoapps/xblock/runtime/runtime.py
index ecfe64adf8..c738044763 100644
--- a/openedx/core/djangoapps/xblock/runtime/runtime.py
+++ b/openedx/core/djangoapps/xblock/runtime/runtime.py
@@ -78,9 +78,7 @@ class XBlockRuntime(RuntimeShim, Runtime):
                 LmsBlockMixin,  # Adds Non-deprecated LMS/Studio functionality
                 XBlockShim,  # Adds deprecated LMS/Studio functionality / backwards compatibility
             ),
-            services={
-                "i18n": ModuleI18nService(),
-            },
+            services={},
             default_class=None,
             select=None,
             id_generator=system.id_generator,
@@ -229,6 +227,9 @@ class XBlockRuntime(RuntimeShim, Runtime):
         elif service_name == "completion":
             context_key = block.scope_ids.usage_id.context_key
             return CompletionService(user=self.user, context_key=context_key)
+        elif service_name == 'i18n':
+            return ModuleI18nService(block=block)
+
         # Check if the XBlockRuntimeSystem wants to handle this:
         service = self.system.get_service(block, service_name)
         # Otherwise, fall back to the base implementation which loads services
```

Essentially, the [ModuleI18nService has to be initialized with the active XBlock](https://github.com/edx-solutions/xblock-drag-and-drop-v2/pull/266), otherwise it can't load that block's translations files.

Reply:
@pomegranited just from looking at it, I know this works, because this is exactly what I found, but I wasn't able to find something similar to this but with modulestore, so I really got confused. 

I am going to add these changes, thank you very much